Code/GraphMol/RascalMCES/RascalMCES.cpp

```diff
@@ -331,14 +331,15 @@ void makeLineGraph(const ROMol &mol, std::vector<std::vector<int>> &adjMatrix) {
 
 // make sure that mol1_bond in mol1 and mol2_bond in mol2 are, if aromatic, in
 // at least one ring that is the same.
-bool checkAromaticRings(const ROMol &mol1,
+bool checkRings(const ROMol &mol1,
                         std::vector<std::string> &mol1RingSmiles,
                         int mol1BondIdx, const ROMol &mol2,
                         std::vector<std::string> &mol2RingSmiles,
-                        int mol2BondIdx) {
+                        int mol2BondIdx,
+                        bool aromaticRingsMatchOnly) {
   auto mol1Bond = mol1.getBondWithIdx(mol1BondIdx);
   auto mol2Bond = mol2.getBondWithIdx(mol2BondIdx);
-  if (!mol1Bond->getIsAromatic() || !mol2Bond->getIsAromatic()) {
+  if (aromaticRingsMatchOnly && (!mol1Bond->getIsAromatic() || !mol2Bond->getIsAromatic())) {
```

Comment:
If aromaticRingsMatchOnly is false, does it make sense at this point to check that the bonds are either both aromatic or both not aromatic?  Given that it later checks that the SMILES for the two rings are the same, it's probably a failure if one bond is aromatic and the other isn't, and that can be rejected early.

Reply:
That sounds pretty sensible, at least for non-aromatic bonds. If they’re aromatic then you could have a problem with different kekule forms, I think.

Reply:
That makes sense; just added a check

Reply:
My bad.  The molecules should not be in kekule form, in which case getBondType() should return AROMATIC for aromatic bonds, so just checking the type should be enough.  getBondLabels relies on this as well, so if it's not true the whole code fails!